keras_cv/layers/preprocessing/mosaic.py

```diff
@@ -210,8 +210,10 @@ def _update_image(self, images, permutation_order, mosaic_centers, index):
             [input_height, input_width],
         )
         # tf.image.crop_and_resize will always output float32, so we need to recast
-        output = tf.cast(output, self.compute_dtype)
-        return tf.squeeze(output)
+        # tf.image.crop_and_resize outputs [num_boxes, crop_height, crop_width, depth]
+        # since num_boxes is always one we only need the first element
+        output = tf.cast(output[0], self.compute_dtype)
```

Comment:
Probably a safer approach than indexing at 0 would be to use 
`tf.squeeze(output, axis=0)`
(This will raise an exception if the 0 axis has dimension > 1)

Reply:
changed it